main.py

```diff
@@ -49,32 +47,62 @@ def init(viewer):
     glNewList(geometry_display_list, GL_COMPILE)
     glPushAttrib(GL_ENABLE_BIT)
     glEnable(GL_LIGHTING)
+
     for t in tilemap.values():
         if t[2] == 1:
             color = np.array((0.5, 0.5, 0.5))
         elif t[2] == 2:
             color = np.array((0.9, 0.9, 0.0))
         elif t[2] == 3:
-            color = np.array((0.9, 0.7, 0.0))
+            color = np.array((0.0, 0.7, 0.0))
         else:
             color = np.array((0.9, 0.1, 0.1))
         glMaterialfv(GL_FRONT_AND_BACK, GL_DIFFUSE, (0.7, 0.7, 0.7))
         glMaterialfv(GL_FRONT_AND_BACK, GL_AMBIENT, color)
         gl_geometry.draw_solid_prism(t[0])
 
-        glDisable(GL_DEPTH_TEST)
-        glDisable(GL_LIGHTING)
+    glDisable(GL_DEPTH_TEST)
+    glDisable(GL_LIGHTING)
+
+    glPopAttrib(GL_ENABLE_BIT)
+    glEndList()
+
+    global normal_display_list
+    normal_display_list = glGenLists(1)
+    glNewList(normal_display_list, GL_COMPILE)
+    glPushAttrib(GL_ENABLE_BIT)
+
+    for t in tilemap.values():
         glColor3f(1, 1, 1)
         gl_geometry.draw_prism_normals(t[0], 100.0)
-        glEnable(GL_DEPTH_TEST)
-        glEnable(GL_LIGHTING)
 
-    glDisable(GL_DEPTH_TEST)
-    glDisable(GL_LIGHTING)
+    glPopAttrib(GL_ENABLE_BIT)
+    glEndList()
+
+    global wire_display_list
+    wire_display_list = glGenLists(1)
+    glNewList(wire_display_list, GL_COMPILE)
+    glPushAttrib(GL_ENABLE_BIT)
+
     glColor3f(1, 1, 1)
     for t in tilemap.values():
         gl_geometry.draw_wire_prism(t[0])
 
+
+    glPopAttrib(GL_ENABLE_BIT)
+    glEndList()
+
+    global sample_display_list
+    sample_display_list = glGenLists(1)
+    glNewList(sample_display_list, GL_COMPILE)
+    glPushAttrib(GL_ENABLE_BIT)
+
+    for t in tilemap.values():
```

Comment:
These samples are colorful triangles inside each fd. This is really helpful for checking whether the reflections are correct.